File: web/src/types/project.ts
export interface Agent {
  id: string;
  name: string;
  type: string;
  description?: string;
  createdAt: Date;
  updatedAt: Date;
}

export interface Project {
  id: string;
  name: string;
  directory: string;
  description?: string;
  agentId?: string;
  agent?: Agent;
  createdAt: Date;
  updatedAt: Date;
}

export interface CreateProject {
  name: string;
  directory: string;
  description?: string;
  agentId?: string;
}

export interface UpdateProject {
  name?: string;
  directory?: string;
  description?: string;
  agentId?: string;
}

File: internal/model/project.go
package model

import (
	"time"
)

type Project struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	Directory   string    `json:"directory"`
	AgentID     *string   `json:"agent_id,omitempty"`
	Agent       *Agent    `json:"agent,omitempty"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type CreateProjectRequest struct {
	Name        string  `json:"name" binding:"required"`
	Description string  `json:"description"`
	Directory   string  `json:"directory" binding:"required"`
	AgentID     *string `json:"agent_id,omitempty"`
}

type UpdateProjectRequest struct {
	Name        string  `json:"name"`
	Description string  `json:"description"`
	Directory   string  `json:"directory"`
	AgentID     *string `json:"agent_id,omitempty"`
}

type ProjectResponse struct {
	ID          string    `json:"id"`
	Name        string    `json:"name"`
	Description string    `json:"description"`
	Directory   string    `json:"directory"`
	AgentID     *string   `json:"agent_id,omitempty"`
	Agent       *Agent    `json:"agent,omitempty"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`
}

type ProjectListResponse struct {
	Projects []ProjectResponse `json:"projects"`
	Total    int64             `json:"total"`
}